tests/test_sync_manager.py

```diff
@@ -98,22 +106,39 @@ def __init__(self, client):
     def clean(self):
         self.sync_manager.clean_datastore(self.sync_manager.find_datastore(
             self.DATASTORE1_NAME, self.DATASTORE1_STATE))
+        self.sync_manager.clean_datastore(self.sync_manager.find_datastore(
+            self.DATASTORE2_NAME, self.DATASTORE2_STATE))
+        self.sync_manager.clean_subscription(
```

Comment:
Yup. That's a good idea. I will make it so that clean dataset also cleans up the subscriptions associated with it.